docker/build/devdeps.ompi.Dockerfile

```diff
@@ -19,8 +19,10 @@ FROM nvidia/cuda:11.8.0-devel-ubuntu22.04
 SHELL ["/bin/bash", "-c"]
 ARG DEBIAN_FRONTEND=noninteractive
 
+ARG TARGETARCH
 ENV CUDA_INSTALL_PREFIX=/usr/local/cuda-11.8
 ENV COMMON_COMPILER_FLAGS="-march=x86-64-v3 -mtune=generic -O2 -pipe"
+ENV COMMON_COMPILER_FLAGS_ARM="-march=native -mtune=generic -O2 -pipe"
```

Comment:
Does this pass the x86 flag to ARM? (But leans on "last one wins" command line processing?)

Reply:
No, it is using one or the other:
Throughout the entire file, you will see the line
```
export common_flags=$([ "$TARGETARCH" == "arm64" ] && echo "$COMMON_COMPILER_FLAGS_ARM" || echo "$COMMON_COMPILER_FLAGS")
```
before any compilation. This sets the command line flags that are passed to either be COMMON_COMPILER_FLAGS if we are not targeting arm64 or COMMON_COMPILER_FLAGS_ARM if we are targeting arm64. 

This is of course still not very general, and -march=native is not what we will want to keep (this would limit the range of possible host processors too much), see also the docs [here](https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html). 
This was merely a quick attempt to see if I can compile for arm at all (it does not yet). The primary goal of the PR is merely to set up the workflows to be able to configure multi-platform builds for the docker images. Actually getting an arm build for Open MPI working is deferred for future work.